GROOVY-11582: remove excess manifest attributes

eric-milles opened the pull request:
I suspect not all of these are still required.  This is a draft to explore removing all or some.  And for any kept, adding some docs and tests.

* `Ant-Version`
* `Bundle-ClassPath`
* `Created-By`
* `DynamicImport-Package`
* `Eclipse-BuddyPolicy`
* `Eclipse-ExtensibleAPI`
* `Extension-Name`
* `Tool`

eric-milles commented:
This is also a work-in-progress for dropping `Ant-Version` and `Created-By` from the jarjar output, which retains those attributes due to use of ant jar task to repackage the jarjar archive.

paulk-asert commented on build-logic/src/main/groovy/org/apache/groovy/gradle/JarJarTask.groovy:
```diff
@@ -169,16 +168,9 @@ class JarJarTask extends DefaultTask {
-                manifest {
-                    // because we don't want to use JDK 1.8.0_91, we don't care and it will
-                    // introduce cache misses
-                    attribute(name: 'Created-By', value: 'Gradle')
```
-1 on deleting this
MANIFEST.MF now contains something like:

> Created-By: 17.0.13+11-LTS (Azul Systems, Inc.)


Instead of:

> Created-By: Gradle

Which, as well as impacting Gradle cache, breaks our reproducible build goal. "Created-By" is already in the `excludedFromManifest` list, so I guess can't really be excluded, hence its forced value.

eric-milles replied:
This should be all set now.  I can promote it from draft or you can do some investigation.  Just let me know.

paulk-asert replied:
+1 for all refactoring changes and removing Ant-Version, Created-By, Extension-Name, Tool. My preference is to keep OSGi changes in a separate PR, ideally with some testing, but at least to make it easier/cleaner for manual testing.

paulk-asert commented:
I merged in the "pre-work" and minor refactoring. Do you want to rebase this against master? Should reduce to just a few lines.

eric-milles commented:
I have rebased, added comments to the `Eclipse-ExtensibleAPI` line, and moved `DynamicImport-Package` and `Eclipse-BuddyPolicy` to main jar only.  I should be able to dig up a jira ticket number for those two from the comments you left on GROOVY-11582.

I still think all 3 can go, but the urgency is not that great.

Just a note, I tried `Main-Class` in the `org.apache.groovy-main` plugin, but it did not end up at the end of the manifest, where I wanted it to be.  That is why I added a project name check to `GroovyLibraryExtension`.
